pkg/kubelet/gpu/nvidia/nvidia_gpu_manager.go

```diff
@@ -101,8 +102,7 @@ func (ngm *nvidiaGPUManager) Start() error {
 	if err := ngm.discoverGPUs(); err != nil {
 		return err
 	}
-	// It's possible that the runtime isn't available now.
-	ngm.allocated = ngm.gpusInUse()
+
```

Comment:
Ok. I suppose things should suss themselves out in the unit & e2e tests. I think this should be helped a bit once checkpoints are added with https://github.com/kubernetes/kubernetes/issues/43240.